### PR TITLE
Add Change History colour configuration

### LINKED ISSUE #12321: Feature Request: Allow setting Change History colors
### Description of the Issue

This isn't a bug report but rather a feature request if possible.

The new "Change History" feature should have an option to change the colors of the changes made and not just have green and red. This would be of great help for those of us who are colorblind and have difficulty telling both colors apart from each other.

### Steps to Reproduce the Issue

1. Open a file
2. Make changes to display the first color
3. Save the file to display the second color

### Debug Information

Notepad++ v8.4.6   (64-bit)
Build time : Sep 25 2022 - 19:51:39
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : C:\_BACKUPS\_NPP
OS Name : Windows 10 Pro (64-bit) 
OS Version : 21H2
OS Build : 19044.2075
Current ANSI codepage : 1252
Plugins : 
    ComparePlugin (2.0.2)
    DSpellCheck (1.4.24)
    JSMinNPP (1.2205)
    mimeTools (2.8)
    NppConverter (4.4)
    NppExport (0.4)
    NppSaveAsAdmin (1.0.211)
    XMLTools (3.1.1.13)
    zoomdisabler_x64 (1.2)

### LINKED ISSUE #13915: Make "Display Change History" customizeable
The color that's used in "Display Change history" is a bright neon orange. This is pretty ugly and I think we should add a feature to customize its color scheme.

### PR DESCRIPTION
**Pull request by mpheath**
fix #12321
fix #13915

### CODE REVIEW

**Comment by chcg** [comment]
This modifies scintilla code. So I think that part should go here https://sourceforge.net/p/scintilla/feature-requests/  if this is a change of general interest.

**Comment by alankilborn** [comment]
@chcg 

>This modifies scintilla code

All files changed by this PR are in PowerEditor folder, so I don't see how this modified Scintilla code?

**Comment by chcg** [comment]
@alankilborn Thanks for the correction. I was wrong.

File: PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp [hunk 1]
```diff
@@ -3003,6 +3003,50 @@ void ScintillaEditView::performGlobalStyles()
...
+	COLORREF changeModifiedfgColor = orange;
+	pStyle = stylers.findByName(TEXT("Change History modified"));
+	if (pStyle)
+	{
+		changeModifiedfgColor = pStyle->_fgColor;
+	}
+	execute(SCI_MARKERSETFORE, SC_MARKNUM_HISTORY_MODIFIED, changeModifiedfgColor);
+	execute(SCI_MARKERSETBACK, SC_MARKNUM_HISTORY_MODIFIED, changeModifiedfgColor);
+	execute(SCI_INDICSETFORE, INDICATOR_HISTORY_MODIFIED_INSERTION, changeModifiedfgColor);
+	execute(SCI_INDICSETFORE, INDICATOR_HISTORY_MODIFIED_DELETION, changeModifiedfgColor);
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:3015]: Why do we need the following lines?
```cpp
	execute(SCI_INDICSETFORE, INDICATOR_HISTORY_MODIFIED_INSERTION, changeModifiedfgColor);
	execute(SCI_INDICSETFORE, INDICATOR_HISTORY_MODIFIED_DELETION, changeModifiedfgColor);
```

**Review by donho — CHANGES_REQUESTED** [review]
Please provide the reason for the following code - it is not necessary normally: 
```cpp
	execute(SCI_INDICSETFORE, INDICATOR_HISTORY_MODIFIED_INSERTION, changeModifiedfgColor);
	execute(SCI_INDICSETFORE, INDICATOR_HISTORY_MODIFIED_DELETION, changeModifiedfgColor);
```
If there's no good reason, then please remove them for all added 4 styles.

File: PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp (hunk 1 above)
Comment (alankilborn) [reply PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:3017]: >With the PythonScript 3 plugin, a user can run editor.setChangeHistory(7) and see markers and indicators of Change History. This makes colouring indicators to match the colouring of markers too important to ignore.

>As for indicators, I consider that code that you question should be included to complete the implementation of colouring Change History in this PR.

No one asked for my advice :-) but I would code for Notepad++ "the way Notepad++ is".
Maybe don't code for it "the way it should be".  :-)

The author already has way too many things to think about (ref. open issue count), so giving him all the contingent possibilities that can be set via PythonScript (or whatever) manipulation of Scintilla is IMO a recipe for getting your change rejected.  If YOU want additional features added to _Change History_ for your own use, just code it all in PS (I do this).

Comment (vinsworldcom) [reply PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:3017]: Not to stoke the fire, but see https://github.com/vinsworldcom/nppChangedLines for a plugin implementation of this feature.

Comment (donho) [reply PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:3017]: @mpheath 
> Change History is both markers and indicators and should remain in sync with colouring which seems like a good reason. What is shown depends on the setting of SCI_SETCHANGEHISTORY and can change in an instant. With the PythonScript 3 plugin, a user can run editor.setChangeHistory(7) and see markers and indicators of Change History. This makes colouring indicators to match the colouring of markers too important to ignore.

That's the valid argument. Thank you for the explanation.

**Review by donho — APPROVED** [review]

**Pull request closed without merging**